lib/index.ts

```diff
@@ -246,6 +265,91 @@ class ProgressReporter {
 		};
 	}
 
+	buildProgress(onProgress: ProgressCallback): ProgressCallback {
+		const progressRenderer = this.renderProgress;
+		let lastStep = 1;
+		let step = 0;
+		let total = 1;
+		let lastPercentage = 0;
+
+		let progressCallback: ProgressCallback = () => void 0;
+
+		return (evt: any) => {
+			let stream;
+			let id;
+			let status;
+			try {
+				({ id, stream, status } = evt);
+				if (id == null) {
+					id = '';
+				}
+				if (status == null) {
+					status = '';
+				}
+				if (stream == null) {
+					stream = '';
+				}
+
+				// Check for a `Step n/total` message on the stream
+				const streamComponents = stream.match(/^Step (\d+)\/(\d+)\s*:\s*(.+)/);
+				if (streamComponents != null) {
+					let instruction = '';
+					[, step, total, instruction] = streamComponents;
```

Comment:
@toochevere had previously fixed a `docker-progress` bug where `total` and other variables were undefined or null: https://github.com/balena-io-modules/docker-progress/pull/56/files#diff-d5a4b3a0309f2337144861084c1014523cced561eb722cb9684c63f5e41e0567R341-R346

Besides undefined or null, I think neither `total` not `step` are allowed to be zero (respectively because of division by zero, and negative value for `lastPercentage`). To be on the safe side, maybe we need something like:
```suggestion
					[, step, total, instruction] = streamComponents;
					step ||= 1;
					total ||= 1;
					instruction ||= '';
```

Reply:
Makes sense